### PR TITLE
BaSP-MR-54: Subscription, post, put and delete methods

### PR DESCRIPTION
**Pull request by Franro212**
Perform the create, delete and modify methods in the subscriptions file.

### CODE REVIEW

File: src/Components/Subscriptions/index.js [hunk 1]
```diff
@@ -1,9 +1,216 @@
...
+    classes: '',
+    member: '',
+    date: ''
+  });
+  const [showForm, setShowForm] = useState(false);
+  const [button, setButton] = useState('');
+
+  const getSubscriptions = async () => {
+    try {
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/subscriptions`, {
```
Comment (Eugeniagassman) [inline src/Components/Subscriptions/index.js]: you need to change here with ${process.env.REACT_APP_API_URL}/api/subscription for vercel deploy

File: src/Components/Subscriptions/index.js [hunk 2]
```diff
@@ -1,9 +1,216 @@
...
+        member: subscription.member ? subscription.member._id : '',
+        date: subscription.date || ''
+      });
+      setShowForm(true);
+    }
+  };
+
+  const deleteSubscriptions = async (id) => {
+    try {
+      await fetch(`${process.env.REACT_APP_API_URL}/subscriptions/${id}`, {
```
Comment (Eugeniagassman) [inline src/Components/Subscriptions/index.js]: same here /api/

File: src/Components/Subscriptions/index.js [hunk 3]
```diff
@@ -1,9 +1,216 @@
...
+  const onchangeInput = (e) => {
+    setCreate({
+      ...create,
+      [e.target.name]: e.target.value
+    });
+  };
+
+  const createSubscription = async () => {
+    try {
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/subscriptions/`, {
```
Comment (Eugeniagassman) [inline src/Components/Subscriptions/index.js]: and here

File: src/Components/Subscriptions/index.js [hunk 4]
```diff
@@ -1,9 +1,216 @@
...
+        alert('Error creating the subscription.');
+      }
+    } catch (error) {
+      alert(error.message);
+    }
+  };
+
+  const updateSubscription = async (id) => {
+    try {
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/subscriptions/${id}`, {
```
Comment (Eugeniagassman) [inline src/Components/Subscriptions/index.js]: here too

File: src/Components/Subscriptions/index.js [hunk 5]
```diff
@@ -1,9 +1,216 @@
...
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(create)
+      });
+      if (response.ok) {
+        const createdSubscription = await response.json();
+        setSubscriptions((currentSubscriptions) => {
+          return [...currentSubscriptions, createdSubscription];
+        });
+        alert('Subscription successfully created.');
```
Comment (Eugeniagassman) [inline src/Components/Subscriptions/index.js]: you should show the messages from the server

File: src/Components/Subscriptions/index.js [hunk 6]
```diff
@@ -1,9 +1,216 @@
...
+        setShowForm(false);
+      } else {
+        alert('Error creating the subscription.');
```
Comment (Eugeniagassman) [inline src/Components/Subscriptions/index.js]: you should show the messages from the server

File: src/Components/Subscriptions/index.js [hunk 7]
```diff
@@ -1,9 +1,216 @@
...
+        setSubscriptions((currentSubscriptions) => {
+          return currentSubscriptions.map((subs) => {
+            if (subs._id === updatedSubscription._id) {
+              return updatedSubscription;
+            }
+            return subs;
+          });
+        });
+        setShowForm(false);
+        alert('Subscription updated correctly.');
```
Comment (Eugeniagassman) [inline src/Components/Subscriptions/index.js]: you should show the messages from the server

File: src/Components/Subscriptions/index.js [hunk 8]
```diff
@@ -1,9 +1,216 @@
...
+      } else {
+        alert('Error updating the subscription.');
```
Comment (Eugeniagassman) [inline src/Components/Subscriptions/index.js]: same here

File: src/Components/Subscriptions/index.js (hunk 1 above)
Comment (Franro212) [reply src/Components/Subscriptions/index.js]: done

File: src/Components/Subscriptions/index.js (hunk 2 above)
Comment (Franro212) [reply src/Components/Subscriptions/index.js]: done

File: src/Components/Subscriptions/index.js (hunk 3 above)
Comment (Franro212) [reply src/Components/Subscriptions/index.js]: done

File: src/Components/Subscriptions/index.js (hunk 4 above)
Comment (Franro212) [reply src/Components/Subscriptions/index.js]: done

File: src/Components/Subscriptions/index.js (hunk 5 above)
Comment (Franro212) [reply src/Components/Subscriptions/index.js]: done

File: src/Components/Subscriptions/index.js (hunk 6 above)
Comment (Franro212) [reply src/Components/Subscriptions/index.js]: done

File: src/Components/Subscriptions/index.js (hunk 7 above)
Comment (Franro212) [reply src/Components/Subscriptions/index.js]: done

File: src/Components/Subscriptions/index.js (hunk 8 above)
Comment (Franro212) [reply src/Components/Subscriptions/index.js]: done

**Review by Eugeniagassman — APPROVED** [review]

**Pull request merged**